3rdParty/velocypack/include/velocypack/velocypack-common.h

```diff
@@ -110,9 +110,119 @@ uint64_t fasthash32(void const*, std::size_t, uint32_t);
 #define VELOCYPACK_HASH32(mem, size, seed) fasthash32(mem, size, seed)
 #endif
 
+#ifdef __APPLE__
+#include <libkern/OSByteOrder.h>
+#include <machine/endian.h>
+#elif _WIN32
+#include <stdlib.h>
+#elif __linux__
+#include <endian.h>
+#else
+#pragma messsage("unsupported os or compiler")
```

Comment:
I don't think it is necessary to exclude systems. We can provide a default implementation [like so](https://godbolt.org/z/Wcv34T). This is always correct, without the need to detect what the host endianness is. In particular it compiles perfectly on clang and gcc, i.e. using a single `mov` for little endian and if available a `movbe` for big endian. Only MSVC can not compile this, because [MSVC](https://developercommunity.visualstudio.com/content/problem/917784/msvc-not-using-bswap-when-possible.html). We can make an `ifdef` for them.